pkg/runner/runner.go

```diff
@@ -142,7 +141,16 @@ func (r *Runner) Run(ctx context.Context) error {
 	// NOTE: This MUST happen after we save the run config to avoid storing
 	// the secrets in the state store.
 	if len(r.Config.Secrets) > 0 {
-		cfg := config.GetConfig()
+		// Create a manager for config access
+		manager, err := client.NewManager(ctx)
```

Comment:
It would be nice to have a helper method which combines `NewManager` and `GetConfig` as a single operation.